packages/workspace/src/migrations/update-16-0-0/move-workspace-generators-to-local-plugin.ts

```diff
@@ -74,12 +86,71 @@ function collectAndMoveGenerators(tree: Tree, destinationProjectRoot: string) {
         schema: `./src/generators/${joinPathFragments(c, 'schema.json')}`,
         description: schema.description ?? `Generator ${c}`,
       };
-      tree.rename(childDir, joinPathFragments(destinationDir, c));
+      moveFilesInDirectory(
+        tree,
+        childDir,
+        joinPathFragments(destinationDir, c)
+      );
     }
   }
   return generators;
 }
 
+function moveFilesInDirectory(tree: Tree, source: string, destination: string) {
```

Comment:
I think we should add some unit tests for this.

Reply:
Yeah, thats fair - I tested it locally but will get some unit tests in.